### ISSUE TITLE
Question / feature request: Named one-line annotation for python2

### ISSUE THREAD

**Issue by amueller** [python/mypy#1963]
So, this might be asking for an addition to pep484.
Basically, I don't like the two formats that are proposed / implemented for python2.
I will probably have to live with python2 for quite some while (as a maintainer).

The problem with the one-line syntax is that it doesn't allow automatic type inference from default values, and that it doesn't provide a correspondence between the arguments and the types (if you have 10 arguments, you have to count which corresponds to which).

The problem with the multi-line syntax is that it is quite intrusive and very long (in an editor).

It would be great to fix the issues with the one-line version by something like this:
Current multi-line:
```python
def send_email(address,     # type: Union[str, List[str]]
               sender,      # type: str
               cc,          # type: Optional[List[str]]
               bcc,         # type: Optional[List[str]]
               subject='',
               body=None    # type: List[str]
               ):
    # type: (...) -> bool
    """Send an email message.  Return True iff successful."""
    <code>
```
something like this would be great:
```python
def send_email(address,  sender,  cc, bcc,  subject='', body=None):
    # type: (address: Union[str, List[str]], sender: str, cc: Optional[List[str]],
    #    bcc: Optional[List[str]], body: List[str]) -> bool
    """Send an email message.  Return True iff successful."""
    <code>
```
So basically just have the Python3 syntax in a comment, but without the default values. Instead of 7 lines we now have three, where one is the original python2 line and two are legible type annotations, and subject is inferred.

Would this be possible to implement from a technical point? Does that need a pep?

**Comment by gvanrossum** [comment]
This feature request should first be directed to the https://github.com/python/typing/ repo. But I can already tell you what my answer will be: we already have enough ways to write type annotations, we don't want to increase the burden for type checker implementers even more.

**Comment by amueller** [comment]
That seems reasonable, thanks for the reply.

On Jul 29, 2016 16:58, "Guido van Rossum" <notifications@github.com> wrote:

Closed #1963 <https://github.com/python/mypy/issues/1963>.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/python/mypy/issues/1963#event-739721463>, or mute the
thread
